### ISSUE TITLE
Promote aws-support-spring-boot-starter to TLP?

### ISSUE THREAD

**Issue by kinow** [zalando-incubator/automata-misc#7]
Hi, the aws-support-spring-boot-starter project is quite useful, and an excellent use case would be to combine it plus the zalando/spring-cloud-config-aws-kms project. This way users wouldn't be required to configure anything in their projects, given they have the settings in their AWS environment through IAM policies/grants/roles/etc.

Though it can be used as-is, right now, I think it could be released as a component by itself? This would be kind like a promotion to a top level project, or self-contained component, under the zalando GitHub organization.

Any thoughts on this?

ps: thanks for the fantastic Open Source suite of tools, libraries, projects, and also great number of people around them

**Comment by LappleApple** [comment]
Hi Bruno -- thanks for your kind words. I like your idea. Would you be willing to help us make it happen?

**Comment by kinow** [comment]
Sure would be @LappleApple 

Happy to help sending pull requests, writing some code, tests, or testing things locally and providing feedback here.

What about the following list of steps?

- [ ] Add a license
- [ ] Add a simple README with a short description of the module, and point to the license file
- [ ] Update pom.xml, source code packages, tests, etc
- [ ] Move it under the zalando organization
- [ ] Cut a release to Maven Central repository

Cheers
Bruno

**Comment by LappleApple** [comment]
@kinow With https://github.com/zalando/spring-cloud-config-aws-kms/ already in the org, I'm wondering if we could use and update that repo. But all your other suggestions +1. There are a few other basic usability enhancements we'd want to do, as per our [how-to](https://github.com/zalando/zalando-howto-open-source#what-makes-a-project-open-source).

**Comment by kinow** [comment]
@LappleApple,

>@kinow With https://github.com/zalando/spring-cloud-config-aws-kms/ already in the org, I'm wondering if we could use and update that repo.

Not really. This repository, automata-misc, has a few Maven modules/children projects. My proposal is to make the aws-support-spring-boot-starter module a standalone project. This way we can release it and promote from the incubator (and others like myself can use it too :-) ). Hope that makes sense.

>Zalando is having our annual Hack Week next week and maybe they will want to offer this project as a HW opportunity.

Great, thanks for CC'ing others. Will keep an eye for any interesting blog post about it. Quite interested in the Zalando STUPS project, as we are building something similar but in smaller proportions too. Might ping you guys in some other repositories too :-)

Thanks
Bruno

**Comment by LappleApple** [comment]
@kinow Wait, I think I misunderstood -- in your original note you're suggesting using the aws-support-spring-boot-starter with spring-cloud-config-aws-kms, not combining them, right?

**Comment by kinow** [comment]
That's correct @LappleApple 

Right now, if you are only decrypting data in your application, and you are using the KMS library, the only thing you need to specify in your bootstrap.yml is the aws.region property.

By **using** (sorry for the confusion) it with the spring-boot-starter, users won't have to specify anything at all, besides the Maven dependency.

It is quite handy for our case, as we have a few developer teams, that we need to make sure i) will use the KMS encryption, ii) will set up the project themselves, and iii) help them troubleshooting build/deployment issues. So the less they have to worry about while setting up the project the merrier :)

**Comment by LappleApple** [comment]
@kinow OK, all clear now. Still waiting for @jbellmann @harti2006 and others to chime in here -- guys?

**Comment by duergner** [comment]
Sounds like a pretty straight forward plan. I assume I could invest some time either later this week or during HackWeek to contribute.

In order to have it more in line with the rest of the Spring Boot starter packages we should rename it to something like `spring-boot-starter-aws-environment` I'd suggest.

**Comment by harti2006** [comment]
According to [Spring's naming conventions for custom starters](http://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-developing-auto-configuration.html#boot-features-custom-starter-naming) it should probably be called `aws-environment-spring-boot-starter`

**Comment by duergner** [comment]
You're absolutely right - didn't remember the different naming strategy for non-official starter stuff.

**Comment by LappleApple** [comment]
I caught @harti2006 in our work kitchen yesterday -- seems @duergner and @kinow are most available to work on this during this week. Does that make sense to everyone?

**Comment by duergner** [comment]
@kinow happy to finally announce the first release of AWS Support Spring Boot Starter as TLP.

You find find it here on [GitHub](https://github.com/zalando-incubator/aws-support-spring-boot-starter) and the artefacts are available on [Maven Central](https://repo1.maven.org/maven2/org/zalando/aws-support-spring-boot-starter/1.0.0/aws-support-spring-boot-starter-1.0.0.jar)

**Comment by kinow** [comment]
Thanks a lot everyone! Will start using it in a couple of weeks once I'm back from vacation.

**Comment by LappleApple** [comment]
@kinow That's great to hear. Thanks, everyone. :)
